pkg/instancetype/apply/vm.go

```diff
@@ -23,26 +23,32 @@ import (
 
 	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
 	virtv1 "kubevirt.io/api/core/v1"
-
-	"kubevirt.io/kubevirt/pkg/instancetype/find"
-	preferenceFind "kubevirt.io/kubevirt/pkg/instancetype/preference/find"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
 )
 
-type VMApplier struct {
-	vmiApplier         *VMIApplier
-	instancetypeFinder *find.SpecFinder
-	preferenceFinder   *preferenceFind.SpecFinder
+type specFinder interface {
+	Find(*virtv1.VirtualMachine) (*v1beta1.VirtualMachineInstancetypeSpec, error)
+}
+
+type preferenceSpecFinder interface {
+	Find(*virtv1.VirtualMachine) (*v1beta1.VirtualMachinePreferenceSpec, error)
+}
+
+type vmApplier struct {
+	vmiApplier         *vmiApplier
+	instancetypeFinder specFinder
+	preferenceFinder   preferenceSpecFinder
 }
 
-func NewVMApplier(instancetypeFinder *find.SpecFinder, preferenceFinder *preferenceFind.SpecFinder) *VMApplier {
-	return &VMApplier{
+func NewVMApplier(instancetypeFinder specFinder, preferenceFinder preferenceSpecFinder) *vmApplier {
```

Comment:
You should not return a private type / concrete implementatiton but instead also return an interface.

Reply:
Why? The consumer defines the interface not the implementation? What your suggesting is the instancetypeMethods pattern I'm trying to break away from.

Reply:
Good point, @lyarwood. 
TIL that public methods can be called on private structs even from outside of the package where they are defined.

Reply:
Hm, you might be right. GoLand's linter complains about this with `Exported function with the unexported return type` though. Is there anything we're missing?

Reply:
Is goland using the deprecated golint by any chance?

staticheck through golangci-lint doesn't have an issue with this and at the moment it's the only linter I care about :)

Reply:
Is it possible to see from the outside which methods and unexported type implements? I would assume that you can't see anything on an unexported type? There might also be issues with documentation with this approach.

See discussion here https://github.com/golang/lint/issues/210